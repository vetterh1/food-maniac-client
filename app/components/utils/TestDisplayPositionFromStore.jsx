import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import * as CoordinatesActions from '../../actions/CoordinatesActions';

export class TestDisplayPositionFromStore extends React.Component {
  // Save in Redux store the fake locations
  dispatchAction = (latitude, longitude) => {
    const { dispatch } = this.props;  // Injected by react-redux
    const action = CoordinatesActions.setCurrentLocation(latitude, longitude, false);
    dispatch(action);
  };

  clickRandom = (e) => {
    e.preventDefault();
    this.dispatchAction(Math.floor(Math.random() * 180) - 90, Math.floor(Math.random() * 180) - 90);
    this.props.onClick();
  };

  clickStrasbourg = (e) => {
    e.preventDefault();
    this.dispatchAction(48.5876, 7.7408);
    this.props.onClick();
  };

  clickLille = (e) => {
    e.preventDefault();
    this.dispatchAction(50.6405856, 3.064508);
    this.props.onClick();
  };

  render = () => {
    return (
      <div>
        <ul>
          <li><Button onClick={this.clickRandom}>Random</Button></li>
          <li><Button onClick={this.clickStrasbourg}>Strasbourg</Button></li>
          <li><Button onClick={this.clickLille}>Lille</Button></li>
        </ul>
      </div>
    );
  }
}

TestDisplayPositionFromStore.propTypes = {
  onClick: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default connect()(TestDisplayPositionFromStore);
