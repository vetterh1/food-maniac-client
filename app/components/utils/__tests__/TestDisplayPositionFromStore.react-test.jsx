
import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import { TestDisplayPositionFromStore } from '../TestDisplayPositionFromStore';

describe('TestDisplayPositionFromStore', () => {
  // Minimal props to render component
  const minProps = {
    dispatch: () => {},
    onClick: () => {},
  };

  it('should render without throwing an error and match its empty snapshot', () => {
    const wrapper = shallow(<TestDisplayPositionFromStore {...minProps} />);
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 3 buttons', () => {
    const wrapper = shallow(<TestDisplayPositionFromStore {...minProps} />);
    const buttons = wrapper.find(Button);
    expect(buttons).toHaveLength(3);
    buttons.first().simulate('click', { preventDefault: () => {} });
  });
});
