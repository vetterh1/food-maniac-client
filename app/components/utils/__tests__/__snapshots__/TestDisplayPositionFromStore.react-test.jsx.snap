// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestDisplayPositionFromStore should render without throwing an error and match its empty snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ul>
        <li>
            <ei
                color="secondary"
                onClick={[Function]}
                tag="button"
            >
                Random
            </ei>
        </li>
        <li>
            <ei
                color="secondary"
                onClick={[Function]}
                tag="button"
            >
                Strasbourg
            </ei>
        </li>
        <li>
            <ei
                color="secondary"
                onClick={[Function]}
                tag="button"
            >
                Lille
            </ei>
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div>
      <ul>
            <li>
                  <ei
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
                  >
                        Random
                  </ei>
            </li>
            <li>
                  <ei
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
                  >
                        Strasbourg
                  </ei>
            </li>
            <li>
                  <ei
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
                  >
                        Lille
                  </ei>
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TestDisplayPositionFromStore
        dispatch={[Function]}
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TestDisplayPositionFromStore {
        "_reactInternalInstance": [Circular],
        "clickLille": [Function],
        "clickRandom": [Function],
        "clickStrasbourg": [Function],
        "context": Object {},
        "dispatchAction": [Function],
        "props": Object {
          "dispatch": [Function],
          "onClick": [Function],
        },
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <ul>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Random
                              </ei>
                    </li>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Strasbourg
                              </ei>
                    </li>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Lille
                              </ei>
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <ul>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Random
                              </ei>
                    </li>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Strasbourg
                              </ei>
                    </li>
                    <li>
                              <ei
                                        color="secondary"
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Lille
                              </ei>
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TestDisplayPositionFromStore
    dispatch={[Function]}
    onClick={[Function]}
/>,
}
`;
